components/style/animation.rs

```diff
@@ -825,6 +825,10 @@ fn interpolate_transform_list(from_list: &[TransformOperation],
                 }
                 (&TransformOperation::Rotate(fx, fy, fz, fa),
                  &TransformOperation::Rotate(tx, ty, tz, ta)) => {
+                    let norm_f = ((fx * fx) + (fy * fy) + (fz * fz)).sqrt();
+                    let norm_t = ((tx * tx) + (ty * ty) + (tz * tz)).sqrt();
+                    let (fx, fy, fz) = (fx / norm_f, fy / norm_f, fz / norm_f);
+                    let (tx, ty, tz) = (tx / norm_t, ty / norm_t, tz / norm_t);
                     if fx == tx && fy == ty && fz == tz {
                         let ia = fa.interpolate(&ta, time).unwrap();
                         result.push(TransformOperation::Rotate(fx, fy, fz, ia));
```

Comment:
It says that, if the direction vectors normalize to be the same thing, just interpolate the angle. It doesn't say what to do with the direction vector in the meantime.

If they're different, it says to do the matrix conversion.